sherpa-onnx/csrc/session.cc

```diff
@@ -54,6 +54,61 @@ static Ort::SessionOptions GetSessionOptionsImpl(int32_t num_threads,
       }
       break;
     }
+    case Provider::kTRT: {
+
+      struct TrtPairs {
+        const char* op_keys;
+        const char* op_values;
+      };
+
+      std::vector<TrtPairs> trt_options = {
+        {"device_id","0"},
+        {"trt_max_workspace_size","2147483648"},
+        {"trt_max_partition_iterations","10"},
+        {"trt_min_subgraph_size","5"},
+        {"trt_fp16_enable","0"},
+        {"trt_detailed_build_log","0"},
+        {"trt_engine_cache_enable","1"},
+        {"trt_engine_cache_path","."},
+        {"trt_timing_cache_enable","1"},
+        {"trt_timing_cache_path","."}
+      };
+      
+      // ToDo : Trt configs
+      // "trt_int8_enable",
+      // "trt_int8_use_native_calibration_table",
+      // "trt_dump_subgraphs",
+
+      std::vector<const char*> option_keys,option_values;
+      for (const TrtPairs& pair : trt_options) {
+        option_keys.emplace_back(pair.op_keys);
+        option_values.emplace_back(pair.op_values);
+      }
+
+      std::vector<std::string> available_providers =
+          Ort::GetAvailableProviders();
+      if (std::find(available_providers.begin(), available_providers.end(),
+                "TensorrtExecutionProvider") != available_providers.end()) {
+        const auto& api = Ort::GetApi();
+
+        OrtTensorRTProviderOptionsV2* tensorrt_options;
+        OrtStatus *statusC = api.CreateTensorRTProviderOptions(
+                                    &tensorrt_options);
+        OrtStatus *statusU = api.UpdateTensorRTProviderOptions(
+                tensorrt_options, option_keys.data(), option_values.data(),
+                option_keys.size());
+
+        if (statusC) { OrtStatusFailure(statusC,&os);}
+        else if(statusU) { OrtStatusFailure(statusU,&os);}
+        else {
+          auto *statusTrt = OrtSessionOptionsAppendExecutionProvider_Tensorrt(
+                              sess_opts, 0);
+          if(statusTrt) { OrtStatusFailure(statusTrt,&os);}
+        }
+
+        api.ReleaseTensorRTProviderOptions(tensorrt_options);
+      }
+    }
```

Comment:
Please add a comment here describing that you omit `break` intentionally. If TensorRT is not available, it tries to use CUDA instead.